### DISCUSSION TITLE
Azure AD Authentication for third-party / customer access.

### DISCUSSION THREAD

**Discussion by wiicode** [requarks/wiki#4159] [Help / Questions]
I have Azure AD Authentication working well for our directory, but I made an assumption that I would be able to authenticate third-party M365 orgs (work/school) since my "Accounts in any organizational directory (Any Azure AD directory - Multitenant)" was ON.  

However, I now believe this was an assumption that was very wrong since third-party logins are greeted with:
`Selected user account does not exist in tenant '<directory name>' and cannot access the application '<ID>' in that tenant. The account needs to be added as an external user in the tenant first. Please use a different account.`

Google auth does not behave like this, I can log third parties in without issue - thus the assumption.

I experimented further with Azure Directory B2C, but gave up at this point. 

I guess the questions I have now are:
- Is it possible to authenticate third-party customers (on Microsoft 365 Work-School accounts) using the Azure AD Authentication plugin?
- Is there another option (I looked at doing Auth0 as a middle-man, but then Azure B2C should be equivalent choice)?

**Comment by wiicode** [comment]
Just bumping this in hopes someone sees it. Still scratching my head.

**Reply by amal-viswa** [reply]
Hey, i'm having trouble while setting up azure ad authentication for wiki.js , I've followed the wiki.js official documentation for azure ad , but still not able to login error . can you guys help me resolve it

error 
![Screenshot 2022-08-23 at 18-12-26 Error](https://user-images.githubusercontent.com/90927394/186367518-444d6f4a-c366-4af3-a656-1d80e5c5b11f.png)
-

**Comment by joeldeteves** [comment]
> Just bumping this in hopes someone sees it. Still scratching my head. 

I know it's quite a long time since you posted this.

I managed to work around it by using the generic Oauth2 auth method and setting that up in Entra as a multi tenant app instead.

Works exactly as described here - external auth with no guest users needed.

Users still have to be added in the dashboard unless you enable self registration.

The process for creating the app in Entra is almost the same - you just use the `/organizations` endpoints in WikiJS instead of the tenant one, along with `/.default` for the scope and `id` for the id claim, `displayName` for the Display Name claim and `mail` for the email claim. The user info endpoint is `https://graph.microsoft.com/v1.0/me`

Make sure the app has the required Graph permissions - eg user.read, email.read, etc.

I couldn't get it to work with the built in AAD option unless I used my own endpoint and created the users as guests. That wasn't ideal however and I opted for the multi tenant setup instead.
